examples/next-openai/app/api/assistant/AssistantResponse.ts

```diff
@@ -0,0 +1,65 @@
+export type AssistantStatus = {
+  status: 'in_progress' | 'complete' | 'failed';
+  information?: string;
+};
+
+export type AssistantMessage = {
+  id: string;
+  role: 'assistant';
+  content: Array<{
+    type: 'text';
+    text: {
+      value: string;
+    };
+  }>;
+};
+
+export function AssistantResponse(
+  process: (stream: {
+    sendStatus: (status: AssistantStatus) => void;
+    sendThreadId: (threadId: string) => void;
+    sendMessage: (message: AssistantMessage) => void;
+  }) => Promise<void>,
+): Response {
+  const stream = new ReadableStream({
+    async start(controller) {
+      const textEncoder = new TextEncoder();
+
+      await process({
+        // TODO send custom data
+
+        sendMessage: (message: AssistantMessage) => {
+          // TODO have a smarter streaming protocol that only sends delta + msg id
+          controller.enqueue(
+            textEncoder.encode(`0: ${JSON.stringify(message)}\n\n`),
+          );
```

Comment:
nit: we should use the same utils we use in stream data

Reply:
👍🏻